src/core/src/index/lca_db.rs

```diff
@@ -0,0 +1,387 @@
+use std::str;
+use std::ffi::{CStr, CString};
+use std::collections::HashMap;
+use std::os::raw::c_char;
+use std::convert::From;
+use std::path::Path;
+
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+use serde::de::Deserializer;
+use serde::ser::{SerializeStruct, Serializer};
+use serde::{Deserialize, Serialize};
+use typed_builder::TypedBuilder;
+
+#[cfg(all(target_arch = "wasm32", target_vendor = "unknown"))]
+use wasm_bindgen::prelude::*;
+
+use crate::index::{Index, Comparable};
+use crate::sketch::minhash::{KmerMinHash, HashFunctions, max_hash_for_scaled};
+use crate::signature::{Signature, SigsTrait};
+use crate::Error;
+use crate::ffi::lca_db::AcceptedLineagePair;
```

Comment:
Bringing things from `crate::ffi` is usually a sign that whatever is defined there should probably be here instead. (Same with `std::ffi` and `std::os::raw::c_char`, they should rarely be used outside `crate::ffi`)